### PR TITLE
build: Add migration_script_verification

### PR DESCRIPTION
**Pull request by johevemi**
## Description

Add a migration script verification step to CI that makes sure that migration scripts have a yyyyMMddhhmm prefix and that the datetime prefix does not lay before any existing scripts or later than the current datetime

## References

* #0000

### CODE REVIEW

File: .github/workflows/ci-databricks.yml [hunk 1]
```diff
@@ -25,6 +25,79 @@ jobs:
...
+          CURRENT_DATETIME=$(TZ=Europe/Paris date +"%Y-%m-%d %H:%M:%S")
+          #Internal Field Seperator
+          IFS=','
+
+          # Get the comma seperated string saved to output current_files
+          CURRENT_FILES="${{ steps.get_current_files.outputs.current_files }}"
+          # Turns comma seperated string in to an array using the IFS
+          read -ra CURRENT_FILES_ARRAY <<<"$CURRENT_FILES"
+
+          # Get the comma seperated string saved to output main_files
```
Comment (BjarkeMeier) [inline .github/workflows/ci-databricks.yml:64]:
```suggestion
          # Get the comma separated string saved to output main_files
```

File: .github/workflows/ci-databricks.yml [hunk 2]
```diff
@@ -25,6 +25,79 @@ jobs:
...
+          echo "::set-output name=main_files::$MAIN_FILES"
+
+      - name: Verify migration script names
+        run: |
+          CURRENT_DATETIME=$(TZ=Europe/Paris date +"%Y-%m-%d %H:%M:%S")
+          #Internal Field Seperator
+          IFS=','
+
+          # Get the comma seperated string saved to output current_files
+          CURRENT_FILES="${{ steps.get_current_files.outputs.current_files }}"
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply .github/workflows/ci-databricks.yml:60]: Food for thought: What is "current"? What we currently have in the system/main? Or what we currently have in the PR changes? "Current" might not be a helpful name.

File: .github/workflows/ci-databricks.yml [hunk 3]
```diff
@@ -25,6 +25,79 @@ jobs:
...
+          # Turns comma seperated string in to an array using the IFS
+          read -ra CURRENT_FILES_ARRAY <<<"$CURRENT_FILES"
+
+          # Get the comma seperated string saved to output main_files
+          MAIN_FILES="${{ steps.get_main_files.outputs.main_files }}"
+          # Turns comma seperated string in to an array using the IFS
+          read -ra MAIN_FILES_ARRAY <<<"$MAIN_FILES"
+
+          # Reverse sort the MAIN_FILES_ARRAY to get the get the file with the latest prefix datetime first
```
[the remark this answers is not in the document]
Comment (johevemi) [reply .github/workflows/ci-databricks.yml:69]: lol, yeah that is a mistake

File: .github/workflows/ci-databricks.yml [hunk 4]
```diff
@@ -25,6 +25,79 @@ jobs:
       ARCHITECTURE: 'x64'
       WHEEL_WORKING_DIRECTORY: './source/databricks'
 
+  migration_scripts_verification:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v2
```
Comment (dstenroejl) [inline .github/workflows/ci-databricks.yml:32]:
```suggestion
        uses: actions/checkout@v3
```

File: .github/workflows/ci-databricks.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (johevemi) [reply .github/workflows/ci-databricks.yml:60]: good idea

File: .github/workflows/ci-databricks.yml [hunk 5]
```diff
@@ -25,6 +25,79 @@ jobs:
...
+
+      - name: Get files from current branch
+        id: get_current_files
+        run: |
+          # Gets all files in migration_scripts folder on the branch being merged excluding the __init__ file and put them into at comma seperated string
+          CURRENT_FILES=$(find ./source/databricks/package/datamigration/migration_scripts -type f -not -name '__init__*' -printf "%f,")
+          echo "::set-output name=current_files::$CURRENT_FILES"
+
+      - name: Checkout main branch
+        uses: actions/checkout@v2
```
Comment (dstenroejl) [inline .github/workflows/ci-databricks.yml:42]:
```suggestion
        uses: actions/checkout@v3
```

**Review by BjarkeMeier — COMMENTED** [review]
Great job. Perhaps we can work a little with then naming. But I'm still a fan of this work 👍

**Review by dstenroejl — COMMENTED** [review]
Suggested changes

File: .github/workflows/ci-databricks.yml (hunk 4 above)
Comment (johevemi) [reply .github/workflows/ci-databricks.yml]: done

File: .github/workflows/ci-databricks.yml (hunk 5 above)
Comment (johevemi) [reply .github/workflows/ci-databricks.yml]: done

File: .github/workflows/ci-databricks.yml [hunk 6]
```diff
@@ -25,6 +25,79 @@ jobs:
...
+        with:
+          ref: main
+
+      - name: Get files from main branch
+        id: get_main_files
+        run: |
+        # Gets all files in migration_scripts folder on the main branch excluding the __init__ file and put them into at comma seperated string
+          MAIN_FILES=$(find ./source/databricks/package/datamigration/migration_scripts -type f -not -name '__init__*' -printf "%f,")
```
Comment (JesperGomes) [inline .github/workflows/ci-databricks.yml:50]: what happens if we change the folder structure? Could we add a check that 'migration_scripts' exists?

Comment (johevemi) [reply .github/workflows/ci-databricks.yml]: I think it would fail the job when it cant find the folder, but i am not sure

Comment (BjarkeMeier) [reply .github/workflows/ci-databricks.yml]: image

**Review by dstenroejl — APPROVED** [review]

**Pull request merged**